src/docs/arch-tiered-storage.dox

```diff
@@ -0,0 +1,439 @@
+be constructed, like \c "object:A-0000000003.wtobj".
+Navigation like this, constructing names to look up in metadata or
+as data handles doesn't happen often, mostly during startup or
+flush_tier. During high performance paths, the \c WT_TIERED struct
+or the \c WT_BM (block manager struct) associated with the Btree have
+everything we need.
+
+A "tiered:" entry (associated with a tiered table) and a "file:" entry
+(associated with a non-tiered table) behave almost identically in
+the WiredTiger system.  In fact, the \c WT_BTREE_PREFIX macro checks
+to see if a URI matches either one of these prefix strings.  The macro basically
+means "does this thing walk and talk like a Btree?".  In both cases,
+the dhandle found with the given name has a \c dhandle->handle that
+points to the open \c WT_BTREE.
+
+The \c "object:" entries do not have separate in-memory data structures
+(there is a \c WT_TIERED_OBJECT defined in \c tiered.h, but it is not used).
+And while \c "table:", \c "file:" and "tiered:" all have data handles
+using those URIs, there are no separate data handles for each object.
+This is probably a good thing, as scaling the number of data handles
+system-wide has been challenging.
+
+The following table summarizes the various relationships.
+
+URI prefix    | struct type         | has dhandle?  |  dhandle notes
+------------- | ------------------- | ------------- | --------------------------------------
+\c table:     | WT_TABLE            | yes           | the dhandle is cast to \c (WT_TABLE *)
+\c file:      | (none)              | yes, but...   | does not relate to a Btree
+\c tiered:    | WT_TIERED, WT_BTREE | yes           | the dhandle is cast to \c (WT_TIERED *), and dhandle->handle is a \c (WT_BTREE *)
+\c tier:      | WT_TIERED_TIERS     | no            | WT_TIERED->tiers[1] is a (WT_TIERED_TIERS *)
+\c object:    | (none)              | no            | no dhandle for objects
+
+@section ts_buckets Storage Sources, Buckets, and Prefixes
+
+The location of objects in the cloud requires some information to be
+stored: the cloud provider, the bucket used, and an id to reference
+any needed credentials in a key management system. When a connection
+is opened for tiered storage, these are specified and are used in
+the default case.  However, we want the ability to have some tables stored in
+different places, so we need cloud location information to live
+in the \c WT_TIERED object as well. To make this happen, all this
+\a location information is abstracted into a \c WT_BUCKET_STORAGE .
+The connection has a pointer to a \c WT_BUCKET_STORAGE, and each
+\c WT_TIERED does as well.
+
+Among the fields of a \c WT_BUCKET_STORAGE is a pointer to a \c
+WT_STORAGE_SOURCE. The storage source can be thought of as a driver,
+or an abstraction of a cloud provider with operations. WiredTiger has a
+several instances of \c WT_STORAGE_SOURCE, these include the drivers
+for the AWS, GCP, and Azure clouds.  A storage source can be asked to
```

Comment:
Since this is for an internal developer perhaps, it may be worth noting `dir_store` as well. That storage source is often easier to understand. 
ETA: I see you do mention it below. Maybe mention it here and say it will be discussed in more detail below.

Reply:
Fixed.